### ISSUE TITLE
Issue with Power BI Clockify Connector: Unable to Refresh Data in Power BI Service

### ISSUE THREAD

**Issue by safer-lgtm** [OscarValerock/Clockify-PowerBI#17]
I have successfully used the Power BI Clockify connector to load data into Power BI Desktop, and it refreshes without any issues locally. However, I am encountering a problem when trying to refresh the data in Power BI Service.

### Error Details

When attempting to refresh the data in Power BI Service, I receive the following error:

```
Failed to update data source credentials: Web.Contents failed to get contents from 'https://reports.api.clockify.me/v1' (404): Not Found
```
![repprt api false without detail](https://github.com/OscarValerock/Clockify-PowerBI/assets/66846994/c18b18bc-c977-45c3-99d2-5dce925e41b7)

I have ensured that the connector is located where the gateway can access it. Despite this, the error persists, indicating that the data source is inaccessible, a connection timeout occurred, or the data source credentials are invalid.

![grafik](https://github.com/OscarValerock/Clockify-PowerBI/assets/66846994/1e97459b-d693-4654-9434-7a205d40cbd5)
![grafik](https://github.com/OscarValerock/Clockify-PowerBI/assets/66846994/b67df532-0545-45ae-b5b6-050f7460d664)

Additionally, I've tried to bypass the test connection for `reports.api.clockify.me`, but this only leads to another error stating that the data cannot be validated because it was done via a gateway. However, the primary source `api.clockify.me` can be updated without any issues from both the server and desktop.
![grafik](https://github.com/OscarValerock/Clockify-PowerBI/assets/66846994/b923f3db-57c3-434c-b3bf-28f9b71c4fb7)

### Request for Support

I am seeking assistance to resolve this issue. Is there something I am missing in the configuration for the Power BI Service? Any guidance or troubleshooting steps would be greatly appreciated.

Thank you for your support!

**Comment by TinaCastillo09** [comment]
Hi,

Did you manage to work this one out? I am getting the same error.

**Comment by safer-lgtm** [comment]
Try using the absolute path instead of the relative path and skip connection for the data source reports.api.clockify.me/v1 in PBI Service.

**Comment by OscarValerock** [comment]
Guys, @TinaCastillo09 @safer-lgtm , I feel you are mixing connection possibilities. The repository gives two options: through a custom connector and with a PBIT (Template) file; as you are referring to the error coming from "https://reports.api.clockify.me/v1" I believe you are using the PBIT template file and not the custom MEZ connector. 

- Custom connector (MEZ): Requires gateway, when setting the gateway connection skip the connection test. 
- PBIT: Does not require a gateway, but when configuring data sources in the desktop and the service, set both connections to "organizational" and skip the connection test in the service

I only tested both, and they work fine when refreshing from the service: 

![image](https://github.com/OscarValerock/Clockify-PowerBI/assets/60475729/5c39c01a-9b00-4cdf-8180-7335ab53ba64)

**Comment by OscarValerock** [comment]
Hi all, @safer-lgtm @TinaCastillo09 Please let me know if this video clarifies the issue. 

[

https://github.com/OscarValerock/Clockify-PowerBI/assets/60475729/f9aaacdc-5c4c-4f26-bdea-53c849905f8d

](url)

**Comment by safer-lgtm** [comment]
Thank you @OscarValerock, for your support and especially the video.

I can now refresh the dataset in Power BI Service in 1-2 minutes! Amazing! In my own solution, it used to take 15-20 minutes to refresh the dataset. By using the absolute path and skipping the connection test, I noticed that the privacy levels in Power BI Service are different from those in Power BI Desktop, which were set up to organizational. I'm using PBIT (Template) file.

I created a small report considering Billable Hours, but after the refresh, although it was set to TRUE in Power BI Desktop, I get -1 and 0 instead after the refresh in Power BI Service. Any idea why this might be happening?

![grafik](https://github.com/OscarValerock/Clockify-PowerBI/assets/66846994/3dbacd28-3905-466e-a2fb-0d0a4693c744)


Thank you again for your feedback; I really appreciate it.
